src/utilities/findBreakingChanges.js

```diff
@@ -169,46 +177,69 @@ export function findArgChanges(
 
       oldTypeFields[fieldName].args.forEach(oldArgDef => {
         const newArgs = newTypeFields[fieldName].args;
-        const newTypeArgIndex = newArgs.findIndex(
+        const newArgDef = newArgs.find(
           arg => arg.name === oldArgDef.name
         );
-        const newArgDef = newArgs[newTypeArgIndex];
-
-        const oldArgTypeName = getNamedType(oldArgDef.type);
-        const newArgTypeName = newArgDef ?
-          getNamedType(newArgDef.type) :
-          null;
-
-        if (!oldArgTypeName) {
-          return;
-        }
 
         // Arg not present
-        if (!newArgTypeName) {
+        if (!newArgDef) {
           breakingChanges.push({
             type: BreakingChangeType.ARG_REMOVED,
             description: `${oldType.name}.${fieldName} arg ` +
               `${oldArgDef.name} was removed`,
           });
-
-        // Arg changed type in a breaking way
-        } else if (
-          oldArgTypeName.name !== newArgTypeName.name
-        ) {
-          breakingChanges.push({
-            type: BreakingChangeType.ARG_CHANGED_KIND,
-            description: `${oldType.name}.${fieldName} arg ` +
-              `${oldArgDef.name} has changed type from ` +
-              `${oldArgDef.type.toString()} to ${newArgDef.type.toString()}`,
-          });
-
-        // Arg default value has changed
-        } else if (oldArgDef.defaultValue !== undefined &&
+        } else {
+          let isSafe = false;
+          if (isNamedType(oldArgDef.type)) {
+            isSafe = isNamedType(newArgDef.type) &&
+              getNamedType(oldArgDef.type).name ===
+                getNamedType(newArgDef.type).name;
+          } else if (oldArgDef.type instanceof GraphQLList) {
+            isSafe = newArgDef.type instanceof GraphQLList &&
+              getNamedType(oldArgDef.type) === getNamedType(newArgDef.type);
+          } else if (oldArgDef.type instanceof GraphQLNonNull) {
+            isSafe =
+              (
+                newArgDef.type instanceof GraphQLNonNull &&
+                getNamedType(oldArgDef.type) === getNamedType(newArgDef.type)
+              ) ||
+              // moving from non-null to nullable of the same type is safe for
+              // field args
+              (
+                !(newArgDef.type instanceof GraphQLNonNull) &&
+                isNamedType(newArgDef.type) &&
+                getNamedType(oldArgDef.type).name ===
+                  getNamedType(newArgDef.type).name
```

Comment:
Again, do we need `getNamedType(newArgDef.type)` here only because of Flow? (seeing as we already check `isNamedType` above).